redash/models/__init__.py

```diff
@@ -363,6 +363,7 @@ def get_latest(cls, data_source, query, max_age=0, is_hash=False):
             query = cls.query.filter(
                 cls.query_hash == query_hash,
                 cls.data_source == data_source,
+                cls.db_role == db_role,
```

Comment:
Oh, I just realized that this will not do what you expect when `db_role` is `None` (i.e., user is an admin). Instead of finding the results created by the admin, it will just never find any results. (`col == null` is neither `false` nor `true`). You'll need to check for a `None` value and use `cls.db_role.is_(None)`. I think this would work:

```suggestion
                cls.db_role == db_role if db_role is not None else cls.db_role.is_(None),
```

Reply:
This was a good call. I ended up still hitting some "this isn't quite a boolean" errors so rejiggered the statement a bit, but the core point is well taken :+1: